shell/types/store/dashboard-store.types.ts

```diff
@@ -1,29 +1,161 @@
-export interface OptPaginationSort { field: string, asc: boolean }
+import { NAMESPACE_FILTER_NS_FULL_PREFIX, NAMESPACE_FILTER_P_FULL_PREFIX } from '@shell/utils/namespace-filter';
+
+/**
+ * Sort the pagination result
+ *
+ * For more information see https://github.com/rancher/steve?tab=readme-ov-file#sort
+ */
+export interface OptPaginationSort {
+  /**
+   * Name of field within the object to sort by
+   */
+  field: string,
+  asc: boolean
+}
+
+/**
+ * Filter the pagination result by these specific fields
+ *
+ * For example
+ *
+ * - metadata.name=test
+ * - metadata.namespace!=system
+ *
+ * For more information see https://github.com/rancher/steve?tab=readme-ov-file#query-parameters
+ */
+export class OptPaginationFilterField {
+  /**
+   * Name of field within the object to sort by
+   *
+   * This can be optional for some (projectsornamespaces)
+   */
+  field?: string;
+  value: string;
+  equals: boolean;
+
+  constructor(
+    { field, value, equals = true }:
+    { field?: string; value: string; equals?: boolean; }
+  ) {
+    this.field = field;
+    this.value = value;
+    this.equals = equals;
+  }
+}
+
+/**
+ * Filter the pagination result by a number of fields
+ *
+ * OptPaginationFilter can be used in two ways
+ *
+ * 1) OR'd together
+ *  - a=1 OR b=2 OR c=3
+    - Query Param - filter=a=1,b=2,c=3
+ *  - Object structure - [
+ *      [a,1],[b,2],[c,3]
+ *    ]
+ *
+ * 2) AND'd together
+  *- a=1 AND b=2 AND c=3
+ *  - Query Param - filter=a=1&filter=b=2&filter=c=3
+ *  - Object structure - [
+ *      [a,1]
+ *    ],
+ *    [
+ *      [b,2]
+ *    ],
+ *    [
+ *      [c,3]
+ *    ]
+ *
+ * This structure should give enough flexibility to cover all uses
+ *
+ * For more information see https://github.com/rancher/steve?tab=readme-ov-file#filter
+ */
+export class OptPaginationFilter {// implements OptPaginationFilter
```

Comment:
@codyrancher I don't think i've got the definitions for this and OptPaginationFilterField right. I'm making use of them a lot more in another context (converting the node's list and it's secondary resources over to pagination world) and i'd like to make them easier to use... but will do it in that PR. If there are no objections can we save the review of these until that one?